doc/releases/changelog-dev.md

```diff
@@ -102,8 +102,10 @@
 <h3>Bug fixes</h3>
 
 * Fixed bug where the coefficients where not ordered correctly when summing a `ParametrizedHamiltonian`
-  with other operators.
+  with other operators. Fixed a bug introduced by this fix where `Hamiltonian` and `ParametrizedHamiltonian` 
```

Comment:
Since both of these bug fixes happened in the same release cycle, I think you could just merge the two descriptions into a single sentence. Not blocking though